elastic/kibana
# [Uptime] [Synthetics Integration] add new advanced options

Fixes #109732 

This PR adds UI components for the `ignore_https_errors` and `filter_journeys` options. 

<img width="735" alt="Screen Shot 2021-09-16 at 1 06 06 PM" src="https://user-images.githubusercontent.com/11356435/133655559-0c2f82a7-42c3-4048-8aa9-1a8ef400e5d4.png">

This PR also refactors some of the common options across monitor type for better code reuse, and renames a few files. 

_**Testing**_

**Setting up your e2e testing environment**
Overview: You will need to run a local version of kibana and package registry to test pending changes, but will use `elastic-package` to run a snapshot view of elasticsearch, fleet-server, and elastic-agent.  Since fleet server relies on kibana, we'll also use `elastic-package` to start a snapshot view of kibana exclusively for bootstrapping fleet-server, but will point our local version of kibana to a separate port. So we'll be running two versions of Kibana.
1. Stop all instances of elasticsearch or kibana
2. Checkout this PR. In `kibana.dev.yml` settings, add remove `kibana.index` and `xpack.task_manager.index` to remove legacy multitenancy features. This ensures that both our instances of Kibana can share saved objects related to fleet and agent. Also, add the following keys:
```
xpack.encryptedSavedObjects.encryptionKey: "12345678901234567890123456789012"
xpack.fleet.registryUrl: "http://localhost:8080"
server.port: 5602
```
4. Start your local kibana instance using `env BUILD_TS_REFS_DISABLE=true yarn kbn bootstrap && yarn start`
5. Checkout https://github.com/elastic/integrations/pull/1778. Navigate to `packages/synthetics` in the repo directory. Run `elastic-package clean`, then `elastic-package build`.
6. Update elastic package. It was recently updated to add support for the elastic-agent-complete image which contains support for running synthetics agent via heartbeat. Run `go get github.com/elastic/elastic-package`
7. Run `elastic-package stack up -d -v --services "elasticsearch,fleet-server,elastic-agent,package-registry" --version 8.0.0-SNAPSHOT` in the `packages/synthetics` directory. This will start up elasticsearch, fleet-server, elastic-agent, package registry AND kibana. This is because fleet-server requires kibana for bootstrapping.
8. Change the password to `kibana_system` user `curl -u elastic:changeme -X POST "http://localhost:9200/_security/user/kibana_system/_password?pretty" -H 'Content-Type: application/json' -d' { "password" : "changeme" } '`
9. Navigate to your local instance of kibana at **http://localhost:5602**

**Testing filter tags**
1. Navigate to the synthetics integration at `fleet/integrations/synthetics-0.3.1/add-integration`
2. Add a policy name, then select the browser type from the dropdown
3. Fill out the form with the following options: Zip url: `https://github.com/elastic/synthetics-demo/archive/refs/heads/main.zip`, folder `e-commerce`, params `{
    "url": "https://elastic-synthetics.vercel.app/"
}`
4. Open the browser advanced fields option and add the following option to tags
<img width="726" alt="Screen Shot 2021-09-30 at 10 28 37 AM" src="https://user-images.githubusercontent.com/11356435/135485054-cd92f695-ee8e-4bd9-842c-7dfccd6be4f6.png">
5. Save the integration and navigate to Uptime
6. Verify that only this journey was ran (the name may be different depending on what you named your policy
<img width="483" alt="Screen Shot 2021-09-30 at 11 22 10 AM" src="https://user-images.githubusercontent.com/11356435/135485202-5fecc66b-43c6-4f26-96cf-74a92ba296ed.png">

**Testing filter match**
1. Navigate to the synthetics integration at `fleet/integrations/synthetics-0.3.1/add-integration`
2. Add a policy name, then select the browser type from the dropdown
3. Fill out the form with the following options: Zip url: `https://github.com/elastic/synthetics-demo/archive/refs/heads/main.zip`, folder `e-commerce`, params `{
    "url": "https://elastic-synthetics.vercel.app/"
}`
4. Open the browser advanced fields option and add the following option to the match field
<img width="713" alt="Screen Shot 2021-09-30 at 11 11 33 AM" src="https://user-images.githubusercontent.com/11356435/135485077-b92468b5-d4f3-4c74-bc4a-e87aecd4748e.png">
5. Save the integration and navigate to Uptime
6. Verify that only this journey was ran (the name may be different depending on what you named your policy
<img width="482" alt="Screen Shot 2021-09-30 at 11 22 06 AM" src="https://user-images.githubusercontent.com/11356435/135485203-5c716dc3-c011-496c-baed-871d147c8e5a.png">

**Testing ignore http errors**
1. Navigate to the synthetics integration at `fleet/integrations/synthetics-0.3.1/add-integration`
2. Add a policy name, then select the browser type from the dropdown
3. Fill out the form with the following options: Inline script: 
```
step("load homepage", async () => {
    await page.goto('https://wrong.host.badssl.com/');
});
```
4. Open the browser advanced fields option click ignore http errors
<img width="704" alt="Screen Shot 2021-09-30 at 11 21 38 AM" src="https://user-images.githubusercontent.com/11356435/135485943-eb98430c-313d-4873-a73f-e04a39902da9.png">
5. Save the integration and navigate to Uptime
6. Click on the journey to go to the journey details page
7. Verify that the wrong.host.badssl image appears, indicating that an ssl error was ignored
<img width="1461" alt="Screen Shot 2021-09-30 at 11 33 07 AM" src="https://user-images.githubusercontent.com/11356435/135486146-348a06aa-6a26-4ac3-8ce3-10e51499f14d.png">
